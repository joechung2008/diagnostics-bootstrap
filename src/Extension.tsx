import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root flex-grow-1">
      <h2 className="fw-bold">{extensionName}</h2>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
