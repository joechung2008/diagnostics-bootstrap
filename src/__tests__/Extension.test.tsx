import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Extension from "../Extension";

describe("Extension", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Extension
        extensionName="TestExtension"
        config={{ ckey: "cvalue" }}
        stageDefinition={{ stage1: ["step1", "step2"] }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
