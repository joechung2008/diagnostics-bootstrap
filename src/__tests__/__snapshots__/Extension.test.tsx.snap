// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extension > renders correctly 1`] = `
<div
  class="extension-root flex-grow-1"
>
  <h2
    class="fw-bold"
  >
    TestExtension
  </h2>
  <div>
    <h3>
      Configuration
    </h3>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Key
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            ckey
          </td>
          <td>
            cvalue
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div>
    <h3>
      Stage Definitions
    </h3>
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            Key
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            stage1
          </td>
          <td>
            step1, step2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
